send_queue_test.go

```diff
@@ -68,6 +68,28 @@ var _ = Describe("Send Queue", func() {
 		Eventually(done).Should(BeClosed())
 	})
 
+	It("does not block pending send after the queue has stopped running", func() {
+		done := make(chan struct{})
+		go func() {
+			defer GinkgoRecover()
+			q.Run()
+			close(done)
+		}()
+		q.Close()
```

Comment:
Can we use a `PacketConn` that errors here, instead of calling close? That more closely mirrors the actual issue that you discovered, as the session only calls `Close` after calls to `Send` have returned.

Reply:
@marten-seemann Thanks, I updated this to have the `MockConnection` trigger an error on `Write` (how the queue sees that error) -- I'm not sure rigging in an actual `PacketConn` would add much in this context, but let me know.

Reply:
Thank you, that's actually what I meant. Should have been more precise.